festim/hydrogen_transport_problem.py

```diff
@@ -32,11 +32,16 @@ def __init__(
         self.solver_parameters = solver_parameters
         self.exports = exports
 
+        self.dx = None
+        self.ds = None
+        self.function_spaces = []
+
     def initialise(self):
         """Initialise the model. Creates suitable function
         spaces, the subdomains...
         """
 
-        # geometry
-        self.geometry.define_function_space()
-        self.geometry.define_subdomains()
+        self.function_spaces.append(self.geometry.define_function_space())
+        self.geometry.subdomains = self.geometry.mesh.define_measures(
+            self.function_spaces[0]
+        )
```

Comment:
I don't understand. 

`Geometry` has an attribute subdomains of type list of `Subdomain` objects.
Here, you are assigning in this attribute, whatever is returned by `Mesh.define_measures()` which is a `fenicsx.ds` object

There is something wrong here